src/pages/LogOutPreviousUserPage.tsx

```diff
@@ -0,0 +1,60 @@
+import type {StackScreenProps} from '@react-navigation/stack';
+import React, {useEffect} from 'react';
+import {Linking} from 'react-native';
+import {withOnyx} from 'react-native-onyx';
+import type {OnyxEntry} from 'react-native-onyx/lib/types';
+import FullScreenLoadingIndicator from '@components/FullscreenLoadingIndicator';
+import * as SessionUtils from '@libs/SessionUtils';
+import type {AuthScreensParamList} from '@navigation/types';
+import * as SessionActions from '@userActions/Session';
+import ONYXKEYS from '@src/ONYXKEYS';
+import type SCREENS from '@src/SCREENS';
+import type {Session} from '@src/types/onyx';
+
+type LogOutPreviousUserPageOnyxProps = {
+    /** The data about the current session which will be set once the user is authenticated and we return to this component as an AuthScreen */
+    session: OnyxEntry<Session>;
+};
+
+type LogOutPreviousUserPageProps = LogOutPreviousUserPageOnyxProps & StackScreenProps<AuthScreensParamList, typeof SCREENS.TRANSITION_BETWEEN_APPS>;
+
+// This page is responsible for handling transitions from OldDot. Specifically, it logs the current user
+// out if the transition is for another user.
+//
+// This component should not do any other navigation as that handled in App.setUpPoliciesAndNavigate
+function LogOutPreviousUserPage({session, route}: LogOutPreviousUserPageProps) {
+    useEffect(() => {
+        Linking.getInitialURL().then((transitionURL) => {
+            const sessionEmail = session?.email;
+            const isLoggingInAsNewUser = SessionUtils.isLoggingInAsNewUser(transitionURL ?? undefined, sessionEmail);
```

Comment:
NAB: Isn't is kind of weird to do `transitionURL ?? undefined`? Can't we just pass it since it's already a string or null?

Reply:
`SessionUtils.isLoggingInAsNewUser` does not accept `null`. It accepts an optional string (i.e: `string` or `undefined`. So this is technically correct. I could update `SessionUtils.isLoggingInAsNewUser` to accept an optional `string | null`, but I'm not sure that's any better than this.

Reply:
It seems like it's more common to use `undefined` vs `null` in Typescript, for some reason. Thanks for explaining.